command-expansion-server/src/app.ts

```diff
@@ -36,41 +37,77 @@ app.post("/put-dictionary", async (req, res) => {
   console.log(`command-lib generated - path: ${commandDictionaryPath}`);
 
   const sqlExpression = `
-      insert into command_dictionary (command_types, mission, version)
-      values ($1, $2, $3)
-      on conflict (mission, version) do update
-      set command_types = $1
-      returning id;
-    `;
+    insert into command_dictionary (command_types, mission, version)
+    values ($1, $2, $3)
+    on conflict (mission, version) do update
+    set command_types = $1
+    returning id;
+  `;
 
   const { rows } = await db.query(sqlExpression, [
     commandDictionaryPath,
     parsedDictionary.header.mission_name,
     parsedDictionary.header.version,
   ]);
 
-  if (rows.length < 0) {
+  if (rows.length < 1) {
     console.error(`POST /dictionary: No command dictionary was updated in the database`);
-    res.status(400).json({ message: `POST /dictionary: No command dictionary was updated in the database` });
-  } else {
-    const id = rows[0].id;
-    res.status(200).json({ id });
+    res.status(500).send(`POST /dictionary: No command dictionary was updated in the database`);
+    return;
   }
+  const id = rows[0].id;
+  res.status(200).json({ id });
   return;
 });
 
-app.put("/expansion/:activityTypeName", async (req, res) => {
-  res.status(501).send("PUT /expansion: Not implemented");
+app.post('/get-expansion', async (req, res) => {
+  const id = req.body.input.expansionId as string;
+  console.log(`POST /get-expansion: Retrieved expansion from database: id=${id}`);
+
+  const { rows } = await db.query(`
+    SELECT expansion_logic
+    FROM expansion_rules
+    WHERE id = $1;
+  `, [
+    id,
+  ]);
+
+  if (rows.length < 1) {
+    throw new ErrorWithStatusCode(`POST /get-expansion: No expansion with id=${id}`, 404);
+  }
+
+  const expansionLogicTypescriptBase64 = rows[0].expansion_logic;
+  res.status(200).json({
+    typescript: expansionLogicTypescriptBase64,
+  });
```

Comment:
Nah that's a good point.

I guess I was originally thinking that we wanted this `get-expansion` endpoint to serve as a wrapper in case any deserialization escaping logic changed for the `expansion_logic` TypeScript string. But if we're going forward with base64 then I think we can remove this endpoint and just expect the UI/clients to do their own deserialization.

Reply:
Yeah, I started thinking about that as well when we went down the base64 path.

Reply:
Done! Just pushed this